### PR TITLE
Fixed config file being overriden when it's blocked by another application.

### PR DESCRIPTION
**Pull request by drfrag666**

### CODE REVIEW

File: src/common/utility/configfile.cpp [hunk 1]
```diff
@@ -601,10 +604,13 @@ void FConfigFile::LoadConfigFile ()
 	FileReader file;
 	bool succ;
 
-	FileExisted = false;
+	if (!FileExists(PathName))
+		FileExisted = false;
```
Comment (alexey-lysiuk) [inline src/common/utility/configfile.cpp:608]: This should be changed to `FileExisted = FileExists(PathName);`

File: src/common/utility/configfile.cpp [hunk 2]
```diff
@@ -601,10 +604,13 @@ void FConfigFile::LoadConfigFile ()
...
+
 	if (!file.OpenFile (PathName))
 	{
-		return;
+		if (!FileExisted) return;
+		else I_Error ("Could not Open Config file.\n");
```
Comment (alexey-lysiuk) [inline src/common/utility/configfile.cpp:613]: Could you please put `if` condition and statement on different lines? The same for `else` keyword and its statement.
Character case of the first letters in the error message is a bit random as well.

File: src/common/utility/configfile.cpp [hunk 3]
```diff
@@ -39,6 +39,9 @@
 #include "configfile.h"
 #include "files.h"
 
+#include "filesystem.h"
```
Comment (alexey-lysiuk) [inline src/common/utility/configfile.cpp:42]: Is it for `FileExists()` function? If so, this function is defined in `cmdlib.h` file.

**Review by alexey-lysiuk — CHANGES_REQUESTED** [review]

**Comment by drfrag666** [comment]
Done.

**Comment by alexey-lysiuk** [comment]
All speculations about this issue revolve around inability to read existing config file at some point for an unknown reason.
Did you test this change in the given case?

When I removed all permissions from config file, `FileExists()` returns `false` because `_wstat64()` returns zero.
So, the added code doesn't handle the advertised error condition, at least on Windows.

**Comment by drfrag666** [comment]
No but _stat is valid in order to check for file existence in theory. In case it returns -1 it's becouse the file doesn't exist or a parameter is not valid. I wouldn't be surprised if it's another bug in _stat. I could try to force the internal version in LZDoom, i'm using VS 2017 BTW. On the other side ::exits internally uses GetFileAttributesW.

**Comment by drfrag666** [comment]
The problem is real, what happens when you remove the permissions from the file? I assume the config is still overriden, i will do some testing in that situation then to see if it's a bug in _stat or ::exists work.

**Comment by drfrag666** [comment]
Well, i've just confirmed that _stat is broken on windows and the internal version works. I see it's still in GZDoom guarded by #ifdef _USING_V110_SDK71_ and it's _wstat64i32 but now _wstat64 is used instead.

**Pull request closed without merging**